src/JsonApiErrorMiddleware.php

```diff
@@ -7,18 +7,73 @@
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
+use Throwable;
+use WtfPhp\JsonApiErrors\Factories\JsonApiErrorFactory;
 
 class JsonApiErrorMiddleware implements MiddlewareInterface
 {
     private ResponseFactoryInterface $responseFactory;
+    private JsonApiErrorFactory $jsonApiErrorFactory;
+    private JsonApiErrorResponseSchema $jsonApiErrorResponseSchema;
 
-    public function __construct(ResponseFactoryInterface $responseFactory)
-    {
+    public function __construct(
+        ResponseFactoryInterface $responseFactory,
+        JsonApiErrorFactory $jsonApiErrorFactory,
+        JsonApiErrorResponseSchema $jsonApiErrorResponseSchema
+    ) {
         $this->responseFactory = $responseFactory;
+        $this->jsonApiErrorFactory = $jsonApiErrorFactory;
+        $this->jsonApiErrorResponseSchema = $jsonApiErrorResponseSchema;
     }
 
+    // INFO: Currently handles only a single Throwable.
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        return $handler->handle($request);
+        try {
+            return $handler->handle($request);
+        } catch (Throwable $t) {
+            return $this->buildResponse($t);
+        }
+    }
+
+    /**
+     * @param Throwable $t
+     * @return ResponseInterface
+     */
+    private function buildResponse(Throwable $t): ResponseInterface
+    {
+        // TODO NOW: If no prop is set, do I still want to set code and message?
+        // See Test: itShouldHandleAnExceptionWithoutErrorCodeAndMessage
+        $jsonApiError = $this->jsonApiErrorFactory::createFromThrowable($t);
```

Comment:
In my opinion I should always see a title and some information, even if it is a basic one. An exception (even without props) indicates some error, that one should see and be notified about.